python/revng/model/base.py

```diff
@@ -0,0 +1,170 @@
+from dataclasses import dataclass, fields
```

Comment:
It seems to me that the `python/revng/model/` directory contains non-model specific stuff.
It's probably a good time to split model-specific stuff (e.g., `MetaAddress`) from `tuple_tree_generator`'s stuff.

Reply:
I can't think of a better place than the `revng.model` module where `MetaAddress` could be stored. Apart maybe from pushing it up to a `revng.metaaddress` module? IMO it belongs to the model

Reply:
I think there was a misunderstanding. `MetaAddress` belongs to the model. But many others in this file are *not* model-specific but generic to tupletree: `StructBase`, `EnumBase`, `Reference` (in the future).